### PR TITLE
Fix tools azure dep

### PR DESCRIPTION
**Pull request by CMeza99**
While working on #5332 I ran into this error:
```
ERROR: azure-cli-core 2.0.81 has requirement azure-mgmt-resource~=6.0, but you'll have azure-mgmt-resource 4.0.0 which is incompatible.
```

### CODE REVIEW

**Comment by kapilt** [comment]
its an error message but it still installs .. ping @cloud-custodian/azure  afaics the need for this pin was resolved upstream, but we need a code update to handle the latest version?

File: tools/c7n_azure/setup.py [hunk 1]
```diff
@@ -99,7 +96,7 @@
                       "requests",
                       "PyJWT",
                       "c7n>=0.8.45.1",
-                      "azure-cli-core",
+                      "azure-cli-core<2",
```
Comment (kapilt) [inline tools/c7n_azure/setup.py]: why this?

RROR: Could not find a version that satisfies the requirement azure-cli-core<2 (from c7n-azure==0.6.3->-r requirements-dev.txt (line 31)) (from versions: 0.1.0b4, 0.1.0b7, 0.1.0b8, 0.1.0b9, 0.1.0b10, 0.1.0b11, 0.1.1b1, 0.1.1b2, 0.1.1b3, 0.1.2rc1, 0.1.2rc2, 2.0.0, 2.0.1, 2.0.2, 2.0.3, 2.0.4, 2.0.5, 2.0.6, 2.0.7, 2.0.8, 2.0.9, 2.0.10, 2.0.11, 2.0.12, 2.0.13, 2.0.14, 2.0.15, 2.0.16, 2.0.17, 2.0.18, 2.0.20, 2.0.21, 2.0.22, 2.0.23, 2.0.24, 2.0.25, 2.0.26, 2.0.27, 2.0.28, 2.0.29, 2.0.30, 2.0.31, 2.0.32, 2.0.33, 2.0.34, 2.0.35, 2.0.37, 2.0.38, 2.0.40, 2.0.41, 2.0.42, 2.0.43, 2.0.44, 2.0.45, 2.0.46, 2.0.47, 2.0.48, 2.0.49, 2.0.50, 2.0.51, 2.0.52, 2.0.53, 2.0.54, 2.0.55, 2.0.56, 2.0.57, 2.0.58, 2.0.59, 2.0.60, 2.0.61, 2.0.62, 2.0.63, 2.0.64, 2.0.65, 2.0.66, 2.0.67, 2.0.68, 2.0.69, 2.0.70, 2.0.71, 2.0.72, 2.0.73, 2.0.74, 2.0.75, 2.0.76, 2.0.77, 2.0.78, 2.0.79, 2.0.80, 2.0.81)
ERROR: No matching distribution found for azure-cli-core<2 (from c7n-azure==0.6.3->-r requirements-dev.txt (line 31))

**Comment by stefangordon** [comment]
> its an error message but it still installs .. ping @cloud-custodian/azure afaics the need for this pin was resolved upstream, but we need a code update to handle the latest version?

@axis7818 is investigating

**Comment by kapilt** [comment]
fwiw the poetry conversion will in effect correct this as well #5320

**Comment by stefangordon** [comment]
This should be fixed in master now so can abandon this PR.

**Comment by kapilt** [comment]
@stefangordon thanks

**Pull request closed without merging**